File: python/n566_MatrixReshape.py
# https://leetcode-cn.com/problems/reshape-the-matrix/
class Solution:
    def matrixReshape(self, nums, r: int, c: int):
        if nums is None or r * c != len(nums) * len(nums[0]):
            return nums
        for i in range(1, len(nums)):
            nums[0] += nums[1]
            nums.pop(1)
        for j in range(r-1):
            nums.append(nums[j][c:])
            nums[j] = nums[j][0:c]
        return nums 
        
        
if __name__ == "__main__":
    pass          

File: cpp/n566_MatrixReshape.cpp
// https://leetcode-cn.com/problems/reshape-the-matrix/
#include <iostream>
#include <vector>
#include <algorithm>
#include <string>
using namespace std;

class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>>& nums, int r, int c) {
        if (nums.empty() || nums[0].empty()) return nums;
        int R = nums.size();
        int C = nums[0].size();
        int N = R * C;
        if (N % r != 0 || N / r != c) return nums;
        vector<vector<int> > res(r, vector<int>(c, 0));
        for (int i = 0; i < N; ++i) {
            res[i / c][i % c] = nums[i / C][i % C];
        }
        return res;
    }
};

int main(){
    return 0;
}